tests.js

```diff
@@ -16,45 +16,167 @@
 
 class Tests {
   constructor(options) {
-    this.items = options.manifest.items
-        .filter((item) => !options.httpOnly || item.protocol === 'http');
-    this.individualItems = options.manifest.individualItems;
+    this.tests = options.tests;
+    this.endpoints = this.buildEndpoints();
     this.host = options.host;
+    this.httpOnly = options.httpOnly;
   }
 
-  list(after, limit) {
-    let begin; let end;
-    if (after) {
-      const afterURL = new URL(after);
-      const afterIndex = this.items.findIndex((item) => {
-        const itemURL = new URL(`${item.protocol}://${this.host}${item.pathname}`);
-        return itemURL.pathname === afterURL.pathname &&
-            itemURL.protocol === afterURL.protocol;
-      });
+  buildEndpoints() {
+    const endpoints = {};
 
-      if (afterIndex === -1) {
-        throw new Error(`${after} not found in test manifest`);
+    for (const [ident, test] of Object.entries(this.tests)) {
+      for (const scope of test.scope) {
+        let endpoint = '';
+        let httpsOnly = false;
+        switch (scope) {
+          case 'Window':
+            endpoint = '/api/interfaces';
```

Comment:
In a follow-up, maybe `/api/?scope=Window`, modulo naming of "scope" discussed elsewhere. Not only interfaces are being tested any longer, namespaces and attributes/functions on the global scope are too.

Reply:
I just pushed f8934be379e465cdc78a2d514d0f3a6d6c164380 which should provide some assistance for this.  In that commit, I added `?reportToServer=true`, which will tell the website to send the data to the server rather than display it.  It no longer counts on a test being a "main endpoint" to determine if it should report, which works towards eliminating main endpoints altogether.